src/angular/metadataReader.ts

```diff
@@ -55,6 +92,24 @@ export class MetadataReader {
     return new DirectiveMetadata(d, dec, selector);
   }
 
+  protected readPipeMetadata(d: ts.ClassDeclaration, dec: ts.Decorator): DirectiveMetadata {
+    const nameExpression = getDecoratorPropertyInitializer(dec, 'name');
+    const name = nameExpression && isStringLiteralLike(nameExpression) ? nameExpression.text : undefined;
+
+    const pureExpression = getDecoratorPropertyInitializer(dec, 'pure');
+    const isBoolean = pureExpression && isBooleanLiteralLike(pureExpression);
```

Comment:
```suggestion
    const pure = pureExpression && isBooleanLiteralLike(pureExpression) ? pureExpression : undefined;
```

This way you don't need to `cast` below.